### PR TITLE
[TASKMGR] Fix number formatting to respect NLS Number format settings

### PR DESCRIPTION
**Pull request by KRosUser**
## Purpose

TaskMgr currently does not respect local NLS settings and forces 3 digits + comma (US convention) formatting

image

## Proposed changes

Use user defined formatting preference

image

Implementation with a direct reuse of SH_FormatInteger as already implemented in other places ( dll/win32/shell32/dialogs/filedefext.cpp ... )

By opportunity change the design pattern to switch/case as suggested by @katahiromz

### CODE REVIEW

File: base/applications/taskmgr/procpage.c [hunk 1]
```diff
@@ -324,20 +324,60 @@ void ProcessPageOnNotify(WPARAM wParam, LPARAM lParam)
     }
 }
 
-void CommaSeparateNumberString(LPWSTR strNumber, ULONG nMaxCount)
+UINT
+SH_FormatInteger(LONGLONG Num, LPWSTR pwszResult, UINT cchResultMax)
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:328]:
```suggestion
SH_FormatInteger(
    _In_ LONGLONG Num,
    _Out_ LPWSTR pwszResult,
    _In_ UINT cchResultMax)
```

File: base/applications/taskmgr/procpage.c [hunk 2]
```diff
@@ -588,113 +628,95 @@ BOOL PerfDataGetText(ULONG Index, ULONG ColumnIndex, LPTSTR lpText, ULONG nMaxCo
...
     if (ColumnDataHints[ColumnIndex] == COLUMN_GDIOBJECTS)
     {
-        wsprintfW(lpText, L"%lu", PerfDataGetGDIObjectCount(Index));
-        CommaSeparateNumberString(lpText, nMaxCount);
+        SH_FormatInteger(PerfDataGetGDIObjectCount(Index), lpText, nMaxCount);
     }
     if (ColumnDataHints[ColumnIndex] == COLUMN_IOREADS)
     {
         PerfDataGetIOCounters(Index, &iocounters);
         /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.ReadOperationCount); */
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:688]:
```suggestion
```

File: base/applications/taskmgr/procpage.c [hunk 3]
```diff
@@ -588,113 +628,95 @@ BOOL PerfDataGetText(ULONG Index, ULONG ColumnIndex, LPTSTR lpText, ULONG nMaxCo
...
-        _ui64tow(iocounters.ReadOperationCount, lpText, 10);
-        CommaSeparateNumberString(lpText, nMaxCount);
+        SH_FormatInteger(iocounters.ReadOperationCount, lpText, nMaxCount);
     }
     if (ColumnDataHints[ColumnIndex] == COLUMN_IOWRITES)
     {
         PerfDataGetIOCounters(Index, &iocounters);
         /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.WriteOperationCount); */
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:694]:
```suggestion
```

File: base/applications/taskmgr/procpage.c [hunk 4]
```diff
@@ -588,113 +628,95 @@ BOOL PerfDataGetText(ULONG Index, ULONG ColumnIndex, LPTSTR lpText, ULONG nMaxCo
...
-        _ui64tow(iocounters.WriteOperationCount, lpText, 10);
-        CommaSeparateNumberString(lpText, nMaxCount);
+        SH_FormatInteger(iocounters.WriteOperationCount, lpText, nMaxCount);
     }
     if (ColumnDataHints[ColumnIndex] == COLUMN_IOOTHER)
     {
         PerfDataGetIOCounters(Index, &iocounters);
         /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.OtherOperationCount); */
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:700]:
```suggestion
```

File: base/applications/taskmgr/procpage.c [hunk 5]
```diff
@@ -588,113 +628,95 @@ BOOL PerfDataGetText(ULONG Index, ULONG ColumnIndex, LPTSTR lpText, ULONG nMaxCo
...
-        _ui64tow(iocounters.OtherOperationCount, lpText, 10);
-        CommaSeparateNumberString(lpText, nMaxCount);
+        SH_FormatInteger(iocounters.OtherOperationCount, lpText, nMaxCount);
     }
     if (ColumnDataHints[ColumnIndex] == COLUMN_IOREADBYTES)
     {
         PerfDataGetIOCounters(Index, &iocounters);
         /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.ReadTransferCount); */
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:706]:
```suggestion
```

File: base/applications/taskmgr/procpage.c [hunk 6]
```diff
@@ -588,113 +628,95 @@ BOOL PerfDataGetText(ULONG Index, ULONG ColumnIndex, LPTSTR lpText, ULONG nMaxCo
...
-        _ui64tow(iocounters.ReadTransferCount, lpText, 10);
-        CommaSeparateNumberString(lpText, nMaxCount);
+        SH_FormatInteger(iocounters.ReadTransferCount, lpText, nMaxCount);
     }
     if (ColumnDataHints[ColumnIndex] == COLUMN_IOWRITEBYTES)
     {
         PerfDataGetIOCounters(Index, &iocounters);
         /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.WriteTransferCount); */
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:712]:
```suggestion
```

File: base/applications/taskmgr/procpage.c [hunk 7]
```diff
@@ -588,113 +628,95 @@ BOOL PerfDataGetText(ULONG Index, ULONG ColumnIndex, LPTSTR lpText, ULONG nMaxCo
...
-        _ui64tow(iocounters.WriteTransferCount, lpText, 10);
-        CommaSeparateNumberString(lpText, nMaxCount);
+        SH_FormatInteger(iocounters.WriteTransferCount, lpText, nMaxCount);
     }
     if (ColumnDataHints[ColumnIndex] == COLUMN_IOOTHERBYTES)
     {
         PerfDataGetIOCounters(Index, &iocounters);
         /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.OtherTransferCount); */
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:718]:
```suggestion
```

File: base/applications/taskmgr/procpage.c [hunk 8]
```diff
@@ -324,20 +324,60 @@ void ProcessPageOnNotify(WPARAM wParam, LPARAM lParam)
...
-    for (k=0; i<wcslen(strNumber); i++,j++,k++) {
-        if ((k % 3 == 0) && (j > 0))
-            temp[j++] = L',';
-        temp[j] = strNumber[i];
-    }
-    temp[j] = L'\0';
-    wcsncpy(strNumber, temp, nMaxCount);
+    // Print the number in uniform mode
+    WCHAR wszNumber[24];
+    swprintf(wszNumber, L"%I64u", Num);
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:332]:
```suggestion
    StringCchPrintfW(wszNumber, _countof(wszNumber), L"%I64u", Num);
```

File: base/applications/taskmgr/procpage.c [hunk 9]
```diff
@@ -324,20 +324,60 @@ void ProcessPageOnNotify(WPARAM wParam, LPARAM lParam)
     }
 }
 
-void CommaSeparateNumberString(LPWSTR strNumber, ULONG nMaxCount)
+UINT
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:327]:
```suggestion
/*
 * Adapted from SH_FormatInteger in dll/win32/shell32/dialogs/filedefext.cpp.
 */
UINT
```

File: base/applications/taskmgr/precomp.h [hunk 10]
```diff
@@ -19,6 +19,7 @@
 #include <commctrl.h>
 #include <shellapi.h>
 #include <tlhelp32.h>
+#include <winnls.h>
```
Comment (HBelusca) [inline base/applications/taskmgr/precomp.h:22]: Move in-between wingdi and winuser.

**Comment by KRosUser** [comment]
@HBelusca : We're all set !

File: base/applications/taskmgr/procpage.c [hunk 11]
```diff
@@ -588,113 +636,89 @@ BOOL PerfDataGetText(ULONG Index, ULONG ColumnIndex, LPTSTR lpText, ULONG nMaxCo
     }
     if (ColumnDataHints[ColumnIndex] == COLUMN_MEMORYUSAGE)
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:637]: It would be better to use `switch`-`case`.

Comment (KRosUser) [reply base/applications/taskmgr/procpage.c:637]: It's not the purpose of this PR, but you're right. I'll change the design accordingly

Comment (katahiromz) [reply base/applications/taskmgr/procpage.c:637]: for later.

Comment (KRosUser) [reply base/applications/taskmgr/procpage.c:637]: Yes, implementation under test. Commit to follow.

File: base/applications/taskmgr/procpage.c [hunk 12]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
-        CommaSeparateNumberString(lpText, nMaxCount);
-    }
-    if (ColumnDataHints[ColumnIndex] == COLUMN_VIRTUALMEMORYSIZE)
-    {
-        wsprintfW(lpText, L"%lu", PerfDataGetVirtualMemorySizeBytes(Index) / 1024);
-        CommaSeparateNumberString(lpText, nMaxCount);
-        wcscat(lpText, L" K");
-    }
-    if (ColumnDataHints[ColumnIndex] == COLUMN_PAGEDPOOL)
+    switch(ColumnDataHints[ColumnIndex])
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:618]:
```suggestion
    switch (ColumnDataHints[ColumnIndex])
```

File: base/applications/taskmgr/procpage.c [hunk 13]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
-        PerfDataGetIOCounters(Index, &iocounters);
-        /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.OtherTransferCount); */
-        _ui64tow(iocounters.OtherTransferCount, lpText, 10);
-        CommaSeparateNumberString(lpText, nMaxCount);
+        case COLUMN_IMAGENAME:
+            PerfDataGetImageName(Index, lpText, nMaxCount);
+            break;
+
+        case COLUMN_PID:
+            wsprintfW(lpText, L"%lu", PerfDataGetProcessId(Index));
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:625]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetProcessId(Index));
```

File: base/applications/taskmgr/procpage.c [hunk 14]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+        case COLUMN_USERNAME:
+            PerfDataGetUserName(Index, lpText, nMaxCount);
+            break;
+
+        case COLUMN_COMMANDLINE:
+            PerfDataGetCommandLine(Index, lpText, nMaxCount);
+            break;
+
+        case COLUMN_SESSIONID:
+            wsprintfW(lpText, L"%lu", PerfDataGetSessionId(Index));
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:637]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetSessionId(Index));
```

File: base/applications/taskmgr/procpage.c [hunk 15]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            break;
+
+        case COLUMN_CPUUSAGE:
+            wsprintfW(lpText, L"%02lu", PerfDataGetCPUUsage(Index));
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:641]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%02lu", PerfDataGetCPUUsage(Index));
```

File: base/applications/taskmgr/procpage.c [hunk 16]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            break;
+
+        case COLUMN_CPUTIME:
+            time = PerfDataGetCPUTime(Index);
+            gethmsfromlargeint(time, &dwHours, &dwMinutes, &dwSeconds);
+            wsprintfW(lpText, L"%lu:%02lu:%02lu", dwHours, dwMinutes, dwSeconds);
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:647]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu:%02lu:%02lu", dwHours, dwMinutes, dwSeconds);
```

File: base/applications/taskmgr/procpage.c [hunk 17]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            break;
+
+        case COLUMN_MEMORYUSAGE:
+            SH_FormatInteger(PerfDataGetWorkingSetSizeBytes(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
+            break;
+
+        case COLUMN_PEAKMEMORYUSAGE:
+            SH_FormatInteger(PerfDataGetPeakWorkingSetSizeBytes(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:657]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c [hunk 18]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            break;
+
+        case COLUMN_MEMORYUSAGEDELTA:
+            SH_FormatInteger(PerfDataGetWorkingSetSizeDelta(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:662]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c [hunk 19]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            break;
+
+        case COLUMN_VIRTUALMEMORYSIZE:
+            SH_FormatInteger(PerfDataGetVirtualMemorySizeBytes(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
+            break;
+
+        case COLUMN_PAGEDPOOL:
+            SH_FormatInteger(PerfDataGetPagedPoolUsagePages(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:680]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c [hunk 20]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            SH_FormatInteger(PerfDataGetPageFaultCount(Index), lpText, nMaxCount);
+            break;
+
+        case COLUMN_PAGEFAULTSDELTA:
+            SH_FormatInteger(PerfDataGetPageFaultCountDelta(Index), lpText, nMaxCount);
+            break;
+
+        case COLUMN_VIRTUALMEMORYSIZE:
+            SH_FormatInteger(PerfDataGetVirtualMemorySizeBytes(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:675]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c [hunk 21]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            break;
+
+        case COLUMN_PAGEDPOOL:
+            SH_FormatInteger(PerfDataGetPagedPoolUsagePages(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
+            break;
+
+        case COLUMN_NONPAGEDPOOL:
+            SH_FormatInteger(PerfDataGetNonPagedPoolUsagePages(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:685]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

File: base/applications/taskmgr/procpage.c [hunk 22]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            break;
+
+        case COLUMN_BASEPRIORITY:
+            wsprintfW(lpText, L"%lu", PerfDataGetBasePriority(Index));
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:689]:
```suggestion
            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetBasePriority(Index));
```

File: base/applications/taskmgr/procpage.c [hunk 23]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_CPUTIME:
+            time = PerfDataGetCPUTime(Index);
+            gethmsfromlargeint(time, &dwHours, &dwMinutes, &dwSeconds);
+            StringCchPrintfW(lpText, nMaxCount, L"%lu:%02lu:%02lu", dwHours, dwMinutes, dwSeconds);
+            break;
+
+        case COLUMN_MEMORYUSAGE:
+            SH_FormatInteger(PerfDataGetWorkingSetSizeBytes(Index) / 1024, lpText, nMaxCount);
+            wcscat(lpText, L" K");
```
Comment (katahiromz) [inline base/applications/taskmgr/procpage.c:652]:
```suggestion
            StringCchCatW(lpText, nMaxCount, L" K");
```

**Comment by KRosUser** [comment]
@HBelusca : I let you review following changed introduced after comments from @katahiromz and approve/merge if OK for you (pending checks are finally OK).

File: base/applications/taskmgr/procpage.c [hunk 24]
```diff
@@ -324,20 +324,68 @@ void ProcessPageOnNotify(WPARAM wParam, LPARAM lParam)
...
+    else
+        nf.Grouping *= 10;
+
+    /* Format the number */
+    cchResult = GetNumberFormatW(LOCALE_USER_DEFAULT,
+                                0,
+                                wszNumber,
+                                &nf,
+                                pwszResult,
+                                cchResultMax);
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:383]:
```suggestion
    cchResult = GetNumberFormatW(LOCALE_USER_DEFAULT,
                                 0,
                                 wszNumber,
                                 &nf,
                                 pwszResult,
                                 cchResultMax);
```

File: base/applications/taskmgr/procpage.c [hunk 25]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_SESSIONID:
+            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetSessionId(Index));
+            break;
+
+        case COLUMN_CPUUSAGE:
+            StringCchPrintfW(lpText, nMaxCount, L"%02lu", PerfDataGetCPUUsage(Index));
+            break;
+
+        case COLUMN_CPUTIME:
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:644]:
```suggestion
        case COLUMN_CPUTIME:
        {
            LARGE_INTEGER time;
            DWORD dwHours;
            DWORD dwMinutes;
            DWORD dwSeconds;

```

File: base/applications/taskmgr/procpage.c [hunk 26]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
-    if (ColumnDataHints[ColumnIndex] == COLUMN_CPUTIME)
-    {
-        DWORD dwHours;
-        DWORD dwMinutes;
-        DWORD dwSeconds;
+    IO_COUNTERS iocounters;
+    LARGE_INTEGER time;
+    DWORD dwHours;
+    DWORD dwMinutes;
+    DWORD dwSeconds;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:616]:
```suggestion
```

File: base/applications/taskmgr/procpage.c [hunk 27]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_CPUUSAGE:
+            StringCchPrintfW(lpText, nMaxCount, L"%02lu", PerfDataGetCPUUsage(Index));
+            break;
+
+        case COLUMN_CPUTIME:
+            time = PerfDataGetCPUTime(Index);
+            gethmsfromlargeint(time, &dwHours, &dwMinutes, &dwSeconds);
+            StringCchPrintfW(lpText, nMaxCount, L"%lu:%02lu:%02lu", dwHours, dwMinutes, dwSeconds);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:648]:
```suggestion
            break;
        }
```

File: base/applications/taskmgr/procpage.c [hunk 28]
```diff
@@ -561,140 +609,131 @@ void AddProcess(ULONG Index)
...
+            SH_FormatInteger(iocounters.WriteTransferCount, lpText, nMaxCount);
+            break;
+
+        case COLUMN_IOOTHERBYTES:
+            PerfDataGetIOCounters(Index, &iocounters);
+            SH_FormatInteger(iocounters.OtherTransferCount, lpText, nMaxCount);
+            break;
     }
 
     return FALSE;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:739]:
```suggestion
    return TRUE;
```

**Review by HBelusca — APPROVED** [review]

File: base/applications/taskmgr/procpage.c [hunk 29]
```diff
@@ -324,20 +323,68 @@ void ProcessPageOnNotify(WPARAM wParam, LPARAM lParam)
...
 }
 
-void CommaSeparateNumberString(LPWSTR strNumber, ULONG nMaxCount)
+/*
+ * Adapted from SH_FormatInteger in dll/win32/shell32/dialogs/filedefext.cpp.
+ */
+UINT
+SH_FormatInteger(
+    _In_ LONGLONG Num,
+    _Out_ LPWSTR pwszResult,
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:332]:
```suggestion
    _Out_writes_z_(cchResultMax) LPWSTR pwszResult,
```

File: base/applications/taskmgr/procpage.c [hunk 30]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
-    }
-    if (ColumnDataHints[ColumnIndex] == COLUMN_IOOTHERBYTES)
-    {
-        PerfDataGetIOCounters(Index, &iocounters);
-        /* wsprintfW(pnmdi->item.pszText, L"%d", iocounters.OtherTransferCount); */
-        _ui64tow(iocounters.OtherTransferCount, lpText, 10);
-        CommaSeparateNumberString(lpText, nMaxCount);
+        case COLUMN_IMAGENAME:
+            PerfDataGetImageName(Index, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:617]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 31]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_PID:
+            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetProcessId(Index));
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:621]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 32]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_USERNAME:
+            PerfDataGetUserName(Index, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:625]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 33]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_COMMANDLINE:
+            PerfDataGetCommandLine(Index, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:629]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 34]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_SESSIONID:
+            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetSessionId(Index));
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:633]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 35]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_CPUUSAGE:
+            StringCchPrintfW(lpText, nMaxCount, L"%02lu", PerfDataGetCPUUsage(Index));
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:637]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 36]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+        {
+            LARGE_INTEGER time;
+            DWORD dwHours;
+            DWORD dwMinutes;
+            DWORD dwSeconds;
+
+            time = PerfDataGetCPUTime(Index);
+            gethmsfromlargeint(time, &dwHours, &dwMinutes, &dwSeconds);
+            StringCchPrintfW(lpText, nMaxCount, L"%lu:%02lu:%02lu", dwHours, dwMinutes, dwSeconds);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:649]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 37]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+        }
+
+        case COLUMN_MEMORYUSAGE:
+            SH_FormatInteger(PerfDataGetWorkingSetSizeBytes(Index) / 1024, lpText, nMaxCount);
+            StringCchCatW(lpText, nMaxCount, L" K");
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:655]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 38]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_PEAKMEMORYUSAGE:
+            SH_FormatInteger(PerfDataGetPeakWorkingSetSizeBytes(Index) / 1024, lpText, nMaxCount);
+            StringCchCatW(lpText, nMaxCount, L" K");
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:660]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 39]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_MEMORYUSAGEDELTA:
+            SH_FormatInteger(PerfDataGetWorkingSetSizeDelta(Index) / 1024, lpText, nMaxCount);
+            StringCchCatW(lpText, nMaxCount, L" K");
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:665]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 40]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_PAGEFAULTS:
+            SH_FormatInteger(PerfDataGetPageFaultCount(Index), lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:669]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 41]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_PAGEFAULTSDELTA:
+            SH_FormatInteger(PerfDataGetPageFaultCountDelta(Index), lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:673]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 42]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_VIRTUALMEMORYSIZE:
+            SH_FormatInteger(PerfDataGetVirtualMemorySizeBytes(Index) / 1024, lpText, nMaxCount);
+            StringCchCatW(lpText, nMaxCount, L" K");
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:678]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 43]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_PAGEDPOOL:
+            SH_FormatInteger(PerfDataGetPagedPoolUsagePages(Index) / 1024, lpText, nMaxCount);
+            StringCchCatW(lpText, nMaxCount, L" K");
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:683]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 44]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_NONPAGEDPOOL:
+            SH_FormatInteger(PerfDataGetNonPagedPoolUsagePages(Index) / 1024, lpText, nMaxCount);
+            StringCchCatW(lpText, nMaxCount, L" K");
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:688]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 45]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_BASEPRIORITY:
+            StringCchPrintfW(lpText, nMaxCount, L"%lu", PerfDataGetBasePriority(Index));
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:692]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 46]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_HANDLECOUNT:
+            SH_FormatInteger(PerfDataGetHandleCount(Index), lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:696]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 47]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_THREADCOUNT:
+            SH_FormatInteger(PerfDataGetThreadCount(Index), lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:700]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 48]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_USEROBJECTS:
+            SH_FormatInteger(PerfDataGetUSERObjectCount(Index), lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:704]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 49]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_GDIOBJECTS:
+            SH_FormatInteger(PerfDataGetGDIObjectCount(Index), lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:708]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 50]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_IOREADS:
+            PerfDataGetIOCounters(Index, &iocounters);
+            SH_FormatInteger(iocounters.ReadOperationCount, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:713]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 51]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_IOWRITES:
+            PerfDataGetIOCounters(Index, &iocounters);
+            SH_FormatInteger(iocounters.WriteOperationCount, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:718]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 52]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_IOOTHER:
+            PerfDataGetIOCounters(Index, &iocounters);
+            SH_FormatInteger(iocounters.OtherOperationCount, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:723]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 53]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_IOREADBYTES:
+            PerfDataGetIOCounters(Index, &iocounters);
+            SH_FormatInteger(iocounters.ReadTransferCount, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:728]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 54]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_IOWRITEBYTES:
+            PerfDataGetIOCounters(Index, &iocounters);
+            SH_FormatInteger(iocounters.WriteTransferCount, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:733]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 55]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
+
+        case COLUMN_IOOTHERBYTES:
+            PerfDataGetIOCounters(Index, &iocounters);
+            SH_FormatInteger(iocounters.OtherTransferCount, lpText, nMaxCount);
+            break;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:738]:
```suggestion
            return TRUE;
```

File: base/applications/taskmgr/procpage.c [hunk 56]
```diff
@@ -561,143 +608,137 @@ void AddProcess(ULONG Index)
...
     }
 
-    return FALSE;
+    return TRUE;
```
Comment (HBelusca) [inline base/applications/taskmgr/procpage.c:741]:
```suggestion
    return FALSE;
```

**Pull request merged**